### PR TITLE
Handling missing and corrupted s2s token

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1088

### Change description ###
Handling missing and corrupted s2s token

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/notificationservice/service/AuthServiceTest.java [hunk 1]
```diff
@@ -61,9 +61,9 @@ void should_throw_when_invalid_token_received() {
         willThrow(new InvalidTokenException(invalidToken)).given(validator).getServiceName(anyString());
 
         // when
-        InvalidTokenException exception = catchThrowableOfType(
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/notificationservice/service/AuthServiceTest.java:64]: is `InvalidTokenException` still needed?

Comment (doncem) [reply src/test/java/uk/gov/hmcts/reform/notificationservice/service/AuthServiceTest.java:64]: `InvalidTokenException` is library exception. From what I see this PR introduces in house exception usage instead straight from the library

File: src/main/java/uk/gov/hmcts/reform/notificationservice/service/AuthService.java [hunk 2]
```diff
@@ -16,7 +16,11 @@ public String authenticate(String authHeader) {
         if (authHeader == null) {
             throw new UnauthenticatedException("Missing ServiceAuthorization header");
         } else {
-            return authTokenValidator.getServiceName(authHeader);
+            try {
+                return authTokenValidator.getServiceName(authHeader);
+            } catch (Exception ex) {
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/notificationservice/service/AuthService.java:21]: Can it be exact exception instead?

File: src/test/java/uk/gov/hmcts/reform/notificationservice/service/AuthServiceTest.java (hunk 1 above)
Comment (rkondratowicz) [reply src/test/java/uk/gov/hmcts/reform/notificationservice/service/AuthServiceTest.java:64]: if so, then yeah, we can just 'translate' `InvalidTokenException` to http response

**Pull request merged**